ref/src/sw_driver.F90

```diff
@@ -73,15 +93,18 @@ program sw_driver
   ! Read the debug settings from the namelist
   read(nl_unit, nml=debug)
 
+  ! Read the MPI decomposition from the namelist
+  read(nl_unit, nml=decomposition)
+
   ! Get OMP_NUM_THREADS value
   nthreads = omp_get_max_threads()
 
   ! Print out affinity diagnostic information
   ret = print_affinity()
 
-  ! Initialize GPTL if enabled
+  ! Initialize GPTL if enabled or if this is an MPI run
 #ifdef ENABLE_GPTL
-  if (do_profile == 1) then
+  if (do_profile == 1 .or. mpi_run ) then ! GPTL automatically profiles MPI calls
```

Comment:
I'm assuming this is here because if the code is compiled with GPTL enabled, but `do_profile` is off, the auto profiling of the MPI calls will cause things to fail if we haven't initialized GPTL.

I wonder if the more straightforward situation is to always call the `initialize` and `finalize` routines for GPTL if `ENABLE_GPTL` is on whether `do_profile` is on or not.  The `do_profile` would then be used to toggle the actual timers on/off.

Reply:
@christopherwharrop-noaa Yes, that does sound more straightforward. I'll implement it.